app/legacy_lib/create_campaign_gift.rb

```diff
@@ -49,14 +49,16 @@ def self.create(params)
 			end
 		end
 
-		Qx.transaction do
+		successful_gift = Qx.transaction do
 			# are any gifts available?
 			if campaign_gift_option.quantity.nil? || campaign_gift_option.quantity.zero?|| campaign_gift_option.total_gifts < campaign_gift_option.quantity
 				gift = CampaignGift.new params
 				gift.save!
-				return gift
+				gift
 			end
 		end
+
+		return successful_gift if successful_gift
```

Comment:
This one is a little weirder than the other ones but this is removing the return from inside a transaction.

Reply:
Totally forgot to handle this before merging. I've handled this in https://github.com/CommitChange/houdini/pull/1099